qutebrowser/browser/urlmarks.py

```diff
@@ -218,6 +215,24 @@ def quickmark_del(self, name):
         except KeyError:
             raise cmdexc.CommandError("Quickmark '{}' not found!".format(name))
 
+    def get_by_qurl(self, url):
+        """Look up a quickmark by QUrl, returning its name.
+
+        Takes O(n) time, where n is the number of quickmarks.
+        Use a name instead where possible.
+        """
+        if not url.isValid():
+            raise ValueError("Invalid URL: {}".format(url.errorString()))
+        urlstr = url.toString(QUrl.RemovePassword | QUrl.FullyEncoded)
+
+        try:
+            index = list(self.marks.values()).index(urlstr)
+            key = list(self.marks.keys())[index]
```

Comment:
While this sohuld work (as I *think* the order of `.values()` and a subsequent `.keys()` is the same) I think it'd be cleaner to just build a reverse dict:

```python
name_to_url = {name: url for url, name in self.marks.values()}
return name_to_url[urlstr]